<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting clustering trees • clustree</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Plotting clustering trees">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Google analytics --><script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52309538-5', 'auto');
  ga('send', 'pageview');

</script>
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">clustree</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/clustree.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lazappi/clustree">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Plotting clustering trees</h1>
                        <h4 class="author">Luke Zappia</h4>
            
            <h4 class="date">2018-03-23</h4>
          </div>

    
    
<div class="contents">
<section id="what-is-a-clustering-tree" class="level1"><h1>What is a clustering tree?</h1>
<p>Clustering analysis is used in many contexts to group similar samples. One problem when conducting this kind of analysis is how many clusters to use. This is usually controlled by a parameter provided to the clustering algorithm, such as <span class="math inline">\(k\)</span> for <span class="math inline">\(k\)</span>-means clustering.</p>
<p>Statistics designed to help you make this choice typically either compare two clusterings or score a single clustering. A clustering tree is different in that it visualises the relationships between at a range of resolutions.</p>
<p>To build a clustering tree we need to look at how cells move as the clustering resolution is increased. Each cluster forms a node in the tree and edges are constructed by considering the cells in a cluster at a lower resolution (say <span class="math inline">\(k = 2\)</span>) that end up in a cluster at the next highest resolution (say <span class="math inline">\(k = 3\)</span>). By connecting clusters in this way we can see how clusters are related to each other, which are clearly distinct and which are unstable. Extra information about the cells in each node can also be overlaid in order to help make the decision about which resolution to use.</p>
</section><section id="a-simple-example" class="level1"><h1>A simple example</h1>
<p>To demonstrate what a clustering tree looks like we will work through a short example using the well known <code>iris</code> dataset.</p>
<section id="the-data" class="level2"><h2>The data</h2>
<p>The <code>iris</code> dataset consists of measurements (sepal length, sepal width, petal length and petal width) of 150 iris flowers, 50 from each of three species (<em>Iris setosa</em>, <em>Iris versicolor</em> and <em>Iris virginica</em>). For more inforation see <code>?iris</code>. We are going to use a version of this dataset that has already been clustered. Let’s load the data and take a look:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(clustree)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt; Loading required package: ggraph</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Loading required package: ggplot2</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">data</span>(<span class="st">"iris_clusts"</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">head</span>(iris_clusts)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#&gt;   Sepal.Length Sepal.Width Petal.Length Petal.Width Species K1 K2 K3 K4 K5</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#&gt; 1          5.1         3.5          1.4         0.2  setosa  1  1  2  2  4</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#&gt; 2          4.9         3.0          1.4         0.2  setosa  1  1  2  2  4</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#&gt; 3          4.7         3.2          1.3         0.2  setosa  1  1  2  2  4</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#&gt; 4          4.6         3.1          1.5         0.2  setosa  1  1  2  2  4</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#&gt; 5          5.0         3.6          1.4         0.2  setosa  1  1  2  2  4</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#&gt; 6          5.4         3.9          1.7         0.4  setosa  1  1  2  2  4</span></a></code></pre></div>
<p>Here we have a <code>data.frame</code> with the normal <code>iris</code> datasets, the measurements and species, plus some additional columns. These columns contain the cluster assignments from clustering this data using <span class="math inline">\(k\)</span>-means with values ok <span class="math inline">\(k\)</span> from <span class="math inline">\(k = 1\)</span> to <span class="math inline">\(k = 5\)</span>.</p>
</section><section id="plotting-a-tree" class="level2"><h2>Plotting a tree</h2>
<p>This clustering information is all we need to build a clustering tree. Each column must consist of numeric values indicating which cluster each sample has been assigned to. To plot the tree we just pass this information to the <code>clustree</code> function. We also need to specify and <code>prefix</code> string to indicate which columns contain the clusterings.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/iris-plot-1.png" width="672"></p>
<p>We can see that one cluster is very distinct and does not change with the value of <span class="math inline">\(k\)</span>. This is the <em>Iris setosa</em> samples which are very different to the other species. On the other side of the tree we see a single cluster that splits into the two clusters we would expect to see. After this the tree becomes messier and there are node with multiple incoming edges. This is a good indication that we have over clustered the data.</p>
</section><section id="controlling-aesthetics" class="level2"><h2>Controlling aesthetics</h2>
<p>By default the size of each node is related to the number of samples in each cluster and the colour indicates the clustering resolution. Edges are coloured according to the number of samples they represent and the transparency shows the incoming node proportion, the number of samples in the edge divided by the number of samples in the node it points to. We can control these aesthetics by setting them to specific values:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>, <span class="dt">node_colour =</span> <span class="st">"purple"</span>, <span class="dt">node_size =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">         <span class="dt">node_alpha =</span> <span class="fl">0.8</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/iris-aes-static-1.png" width="672"></p>
<p>We can also link these aesthetics to other information we have about the samples. All the additional columns in the dataset are available to be added as attributes to the nodes in our tree. Because each node represents multiple samples as well specifying a column name we need to supply an aggregation function to use. Let’s try colouring the nodes according to the sepal width:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>, <span class="dt">node_colour =</span> <span class="st">"Sepal.Width"</span>,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">         <span class="dt">node_colour_aggr =</span> <span class="st">"mean"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/iris-aes-1.png" width="672"></p>
<p>We can clearly see that the distinct cluster containing the <em>Iris setosa</em> samples has a wider sepal on average compared to the other clusters.</p>
</section><section id="layout" class="level2"><h2>Layout</h2>
<p>By default the tree is drawn using the Reingold-Tilford tree layout algorithm which tries to place nodes below their parents. Alternatively we could use the Sugiyama layout by specifying the <code>layout</code> argument. This algorithm tries to minimise the number of crossing edges and can produce more attractive trees in some cases.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>, <span class="dt">layout =</span> <span class="st">"sugiyama"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/iris-layout-1.png" width="672"></p>
</section></section><section id="clustering-trees-for-scrna-seq-data" class="level1"><h1>Clustering trees for scRNA-seq data</h1>
<p>Clustering has become a core tool for analysing single-cell RNA-sequencing (scRNA-seq) datasets. These datasets contain gene expression measurements from hundreds to hundreds of thousands of cells. Often samples come from complex tissues containing many types of cells and clustering is used to group similar cells together. To make it easier to produce clustering trees for these kinds of datasets we provide interfaces for some of the objects commonly used to analyse scRNA-seq data.</p>
<section id="singlecellexperiment" class="level2"><h2>SingleCellExperiment</h2>
<p>The <code>SingleCellExperiment</code> is one of these common objects, used across a range of Bioconductor packages. Let’s have a look at an example:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"SingleCellExperiment"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; Loading required package: SummarizedExperiment</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; Loading required package: GenomicRanges</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; Loading required package: stats4</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; Loading required package: BiocGenerics</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; Loading required package: parallel</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; The following objects are masked from 'package:parallel':</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt;     anyDuplicated, append, as.data.frame, cbind, colMeans,</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt;     colnames, colSums, do.call, duplicated, eval, evalq, Filter,</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt;     Find, get, grep, grepl, intersect, is.unsorted, lapply,</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt;     lengths, Map, mapply, match, mget, order, paste, pmax,</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt;     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt;     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,</span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt;     tapply, union, unique, unsplit, which, which.max, which.min</span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt; Loading required package: S4Vectors</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt; Attaching package: 'S4Vectors'</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt; The following object is masked from 'package:base':</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt;     expand.grid</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt; Loading required package: IRanges</span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt; Loading required package: GenomeInfoDb</span></a>
<a class="sourceLine" id="cb6-34" data-line-number="34"><span class="co">#&gt; Loading required package: Biobase</span></a>
<a class="sourceLine" id="cb6-35" data-line-number="35"><span class="co">#&gt; Welcome to Bioconductor</span></a>
<a class="sourceLine" id="cb6-36" data-line-number="36"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-37" data-line-number="37"><span class="co">#&gt;     Vignettes contain introductory material; view with</span></a>
<a class="sourceLine" id="cb6-38" data-line-number="38"><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></a>
<a class="sourceLine" id="cb6-39" data-line-number="39"><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></a>
<a class="sourceLine" id="cb6-40" data-line-number="40"><span class="co">#&gt; Loading required package: DelayedArray</span></a>
<a class="sourceLine" id="cb6-41" data-line-number="41"><span class="co">#&gt; Loading required package: matrixStats</span></a>
<a class="sourceLine" id="cb6-42" data-line-number="42"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-43" data-line-number="43"><span class="co">#&gt; Attaching package: 'matrixStats'</span></a>
<a class="sourceLine" id="cb6-44" data-line-number="44"><span class="co">#&gt; The following objects are masked from 'package:Biobase':</span></a>
<a class="sourceLine" id="cb6-45" data-line-number="45"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-46" data-line-number="46"><span class="co">#&gt;     anyMissing, rowMedians</span></a>
<a class="sourceLine" id="cb6-47" data-line-number="47"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-48" data-line-number="48"><span class="co">#&gt; Attaching package: 'DelayedArray'</span></a>
<a class="sourceLine" id="cb6-49" data-line-number="49"><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></a>
<a class="sourceLine" id="cb6-50" data-line-number="50"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-51" data-line-number="51"><span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></a>
<a class="sourceLine" id="cb6-52" data-line-number="52"><span class="co">#&gt; The following object is masked from 'package:base':</span></a>
<a class="sourceLine" id="cb6-53" data-line-number="53"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-54" data-line-number="54"><span class="co">#&gt;     apply</span></a>
<a class="sourceLine" id="cb6-55" data-line-number="55"></a>
<a class="sourceLine" id="cb6-56" data-line-number="56"><span class="kw">data</span>(<span class="st">"sim_sc3"</span>)</a>
<a class="sourceLine" id="cb6-57" data-line-number="57">sim_sc3</a>
<a class="sourceLine" id="cb6-58" data-line-number="58"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb6-59" data-line-number="59"><span class="co">#&gt; dim: 1000 200 </span></a>
<a class="sourceLine" id="cb6-60" data-line-number="60"><span class="co">#&gt; metadata(2): log.exprs.offset sc3</span></a>
<a class="sourceLine" id="cb6-61" data-line-number="61"><span class="co">#&gt; assays(2): counts logcounts</span></a>
<a class="sourceLine" id="cb6-62" data-line-number="62"><span class="co">#&gt; rownames(1000): Gene1 Gene2 ... Gene999 Gene1000</span></a>
<a class="sourceLine" id="cb6-63" data-line-number="63"><span class="co">#&gt; rowData names(2): feature_symbol sc3_gene_filter</span></a>
<a class="sourceLine" id="cb6-64" data-line-number="64"><span class="co">#&gt; colnames(200): Cell1 Cell2 ... Cell199 Cell200</span></a>
<a class="sourceLine" id="cb6-65" data-line-number="65"><span class="co">#&gt; colData names(8): sc3_1_clusters sc3_2_clusters ... sc3_7_clusters</span></a>
<a class="sourceLine" id="cb6-66" data-line-number="66"><span class="co">#&gt;   sc3_8_clusters</span></a>
<a class="sourceLine" id="cb6-67" data-line-number="67"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb6-68" data-line-number="68"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>This is an example simulated scRNA-seq dataset in <code>SingleCellExperiment</code> format that has been clustered using the <code>SC3</code> package. For more details about this dataset and how it was produced see <code><a href="../reference/sim_sc3.html">?sim_sc3</a></code>. The clustering information is held in the <code>coldata</code> slot.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">colData</span>(sim_sc3))</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt; DataFrame with 6 rows and 8 columns</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt;       sc3_1_clusters sc3_2_clusters sc3_3_clusters sc3_4_clusters</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt;             &lt;factor&gt;       &lt;factor&gt;       &lt;factor&gt;       &lt;factor&gt;</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; Cell1              1              2              3              4</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; Cell2              1              2              3              2</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; Cell3              1              1              1              1</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; Cell4              1              1              1              3</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; Cell5              1              2              3              4</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; Cell6              1              2              2              2</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt;       sc3_5_clusters sc3_6_clusters sc3_7_clusters sc3_8_clusters</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt;             &lt;factor&gt;       &lt;factor&gt;       &lt;factor&gt;       &lt;factor&gt;</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; Cell1              3              2              2              3</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt; Cell2              2              1              1              1</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; Cell3              5              4              4              5</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; Cell4              4              3              3              4</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; Cell5              3              2              5              8</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; Cell6              2              1              1              1</span></a></code></pre></div>
<p>We can plot a clustering tree in the same way we did with a <code>data.frame</code>. In this case the clustering column names contain a suffix that needs to be stripped away so we will pass that along as well.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(sim_sc3, <span class="dt">prefix =</span> <span class="st">"sc3_"</span>, <span class="dt">suffix =</span> <span class="st">"_clusters"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/sim_sc3-plot-1.png" width="672"></p>
</section><section id="seurat" class="level2"><h2>Seurat</h2>
<p>The same dataset is also available as <code>seurat</code> object that has been clustered using the <code>Seurat</code> package. See <code><a href="../reference/sim_seurat.html">?sim_seurat</a></code> for details.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"Seurat"</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; Loading required package: cowplot</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; Attaching package: 'cowplot'</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt; The following object is masked from 'package:ggplot2':</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt;     ggsave</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; Loading required package: Matrix</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; Attaching package: 'Matrix'</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; The following object is masked from 'package:S4Vectors':</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;     expand</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="kw">data</span>(<span class="st">"sim_seurat"</span>)</a>
<a class="sourceLine" id="cb9-16" data-line-number="16">sim_seurat</a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; An object of class seurat in project SeuratProject </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt;  1000 genes across 200 samples.</span></a></code></pre></div>
<p>In this case the clustering information is held in the <code>meta.data</code> slot:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">head</span>(sim_seurat<span class="op">@</span>meta.data)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt;       nGene nUMI    orig.ident res.0 res.0.1 res.0.2 res.0.3 res.0.4</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; Cell1   503 4458 SeuratProject     0       0       0       1       1</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; Cell2   549 4722 SeuratProject     0       0       0       1       1</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; Cell3   504 3960 SeuratProject     0       0       0       0       0</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; Cell4   525 4822 SeuratProject     0       0       0       0       0</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; Cell5   492 4058 SeuratProject     0       0       0       1       1</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; Cell6   610 6780 SeuratProject     0       0       0       0       0</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt;       res.0.5 res.0.6 res.0.7 res.0.8 res.0.9 res.1</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; Cell1       1       1       0       0       0     0</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt; Cell2       1       1       0       0       0     0</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; Cell3       0       0       1       1       1     1</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; Cell4       0       0       1       1       1     1</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt; Cell5       1       1       0       0       0     0</span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt; Cell6       0       0       1       1       1     1</span></a></code></pre></div>
<p>Because this object is only used by the <code>Seurat</code> package we can assume the prefix of the clustering columns.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(sim_seurat)</a></code></pre></div>
<p><img src="clustree_files/figure-html/sim_seurat-plot-1.png" width="672"></p>
</section><section id="using-genes-as-aesthetics" class="level2"><h2>Using genes as aesthetics</h2>
<p>As well as being able to use any additional columns for aesthetics we can also use the expression of individual genes. Let’s colour the nodes in the <code>Seurat</code> tree by <code>Gene1</code>. Again we need to supply an aggregation function.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(sim_seurat, <span class="dt">node_colour =</span> <span class="st">"Gene5"</span>, <span class="dt">node_colour_aggr =</span> <span class="st">"median"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/plot-gene-1.png" width="672"></p>
</section></section><section id="modifying-appearence" class="level1"><h1>Modifying appearence</h1>
<p>The <code>clustree</code> function returns a <code>ggplot</code> object which can be modified using functions in the <code>ggplot2</code> or <code>ggraph</code> packages. For example we could change the colour scales used for the nodes and edges:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="st">    </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="st">    </span><span class="kw">scale_edge_color_continuous</span>(<span class="dt">low =</span> <span class="st">"blue"</span>, <span class="dt">high =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#&gt; Scale for 'edge_colour' is already present. Adding another scale for</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="co">#&gt; 'edge_colour', which will replace the existing scale.</span></a></code></pre></div>
<p><img src="clustree_files/figure-html/modify-1.png" width="672"></p>
<section id="legends" class="level2"><h2>Legends</h2>
<p>The way <code>ggplot</code> objects is displayed is relative to the size of the plotting window or output file. While the main plot will always fit sometimes legends will be placed outside the visible area. One solution to this is to simply increase the size of the image. An alternative solution is to turn off some of the legends, either by setting some of the aesthetics to static values or by using the <code>guides</code> function. We could also move them to somewhere they might fit better. For example let’s remove the edge legends and move the rest to the bottom:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="../reference/clustree.html">clustree</a></span>(iris_clusts, <span class="dt">prefix =</span> <span class="st">"K"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>, <span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>)</a></code></pre></div>
<p><img src="clustree_files/figure-html/legends-1.png" width="672"></p>
</section></section>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#what-is-a-clustering-tree">What is a clustering tree?</a></li>
      <li>
<a href="#a-simple-example">A simple example</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-data">The data</a></li>
      <li><a href="#plotting-a-tree">Plotting a tree</a></li>
      <li><a href="#controlling-aesthetics">Controlling aesthetics</a></li>
      <li><a href="#layout">Layout</a></li>
      </ul>
</li>
      <li>
<a href="#clustering-trees-for-scrna-seq-data">Clustering trees for scRNA-seq data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#singlecellexperiment">SingleCellExperiment</a></li>
      <li><a href="#seurat">Seurat</a></li>
      <li><a href="#using-genes-as-aesthetics">Using genes as aesthetics</a></li>
      </ul>
</li>
      <li>
<a href="#modifying-appearence">Modifying appearence</a><ul class="nav nav-pills nav-stacked">
<li><a href="#legends">Legends</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Luke Zappia, Alicia Oshlack.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
